### PR TITLE
measure time spend in readhalf/writehalf

### PR DESCRIPTION
**Pull request by pmnoxx**

### CODE REVIEW

File: utils/near-performance-metrics/src/stats_enabled.rs [hunk 1]
```diff
@@ -15,55 +17,92 @@ use near_rust_allocator_proxy::allocator::{
...
+pub(crate) const SLOW_HALF: Duration = Duration::from_millis(50);
+const MIN_OCCUPANCY_RATIO_THRESHOLD: f64 = 0.03;
 
 pub(crate) static STATS: Lazy<Arc<Mutex<Stats>>> = Lazy::new(|| Arc::new(Mutex::new(Stats::new())));
 pub(crate) static REF_COUNTER: Lazy<Mutex<HashMap<(&'static str, u32), u128>>> =
     Lazy::new(|| Mutex::new(HashMap::new()));
 
+thread_local! {
+    pub static INITIALIZED: RefCell<u32> = RefCell::new(0);
+    pub(crate) static LOCAL_STATS: Lazy<Arc<Mutex<ThreadStats>>> = Lazy::new(|| Arc::new(Mutex::new(ThreadStats::new())));
```
Comment (matklad) [inline utils/near-performance-metrics/src/stats_enabled.rs:39]: Lazy already tracks "is initialized?" state, so just the following might work:

```
pub(crate) static LOCAL_STATS: Lazy<Arc<Mutex<ThreadStats>>> = Lazy::new(|| { 
    let res = Arc::new(Mutex::new(ThreadStats::new()));
    add_to_global_stats(&res);
    res
});
```

**Comment by bowenwang1996** [comment]
@pmnoxx do we still need this PR

**Comment by pmnoxx** [comment]
> @pmnoxx do we still need this PR

 We discussed about adding performance statistics to FramedReader/FramedWriter as well as bandwidth stats. I'll have to rewrite the change once we upgrade new tokio.

**Comment by bowenwang1996** [comment]
@pmnoxx do we still need this PR?

**Comment by pmnoxx** [comment]
I'll close this, as it's no longer needed.

**Pull request closed without merging**